ros/angel_system_nodes/angel_system_nodes/spatial_mapper.py

```diff
@@ -80,38 +80,35 @@ def __init__(self):
 
         self.poses = []
 
-        # default image dimensions used if we fail to get the current image size
-        self.image_height = 720
-        self.image_width = 1280
-
-        # attempt to get the current image size from the object detector
+        # setup the image size query client and make sure the service is running
         self.image_size_client = self.create_client(QueryImageSize, 'query_image_size')
         while not self.image_size_client.wait_for_service(timeout_sec=1.0):
             log.info("Waiting for image size service...")
 
+        # send image size queries unti we get a valid response
         r = self.send_image_size_request()
         log.info(f"Image size response {r}")
-        if r.image_width <= 0 or r.image_height <= 0:
+        while (r.image_width <= 0 or r.image_height <= 0):
             log.warn("Invalid image dimensions."
-                     + "Make sure the object detector node is running and receiving frames")
-            log.warn(f"Using default image dimensions {self.image_width}x{self.image_height}")
-        else:
-            self.image_height = r.image_height
-            self.image_width = r.image_width
+                     + " Make sure the image converter node is running and receiving frames")
 
-        log = self.get_logger()
-        #log.set_level(10)
+            r = self.send_image_size_request()
+            time.sleep(1)
```

Comment:
I presume you want to sleep *before* sending out the request again.

Reply:
Good catch!